### ISSUE TITLE
How can I find it ?

### ISSUE THREAD

**Issue by Shine6Z** [Ultimaker/UraniumExampleOutputDevicePlugin#2]
[https://github.com/Ultimaker/Uranium/blob/master/create_plugin.py](url)

**Comment by Ghostkeeper** [comment]
Sorry, that documentation is outdated. To create a plug-in package to publish, upload it as a "plug-in package" to contribute.ultimaker.com.

**Comment by Shine6Z** [comment]
But I'd like to see a complete example of the plugin code.
Can I find it here?

**Comment by Ghostkeeper** [comment]
No, as it's not the way to create plug-ins any more we're not supporting that.

I secretly ported it to CMake though to be able to release my plug-ins on Github as well. For an example, I recommend you take a look at the Settings Guide releases, which are .curapackage files: https://github.com/Ghostkeeper/SettingsGuide/releases
You can also upload them to contribute.ultimaker.com and then before submitting it downloading a .curapackage for a certain SDK version again. You can then inspect the file.

You're probably best off just mimicking the structure of those .curapackage files. They are just .zip files with a bit of metadata and then the actual plug-in under `files/plugins`.

My CMake script that generates that file is here:

https://github.com/Ghostkeeper/SettingsGuide/blob/8c55f0ea7f32c7e21bfe4bbe87d0f296cb1f9c24/CMakeLists.txt#L128-L161

Essentially that just takes a fixed `_rels` and `[Content_Types].xml` file, generates the correct `package.json` with the version number and such, and then copies in all of the files that need to be packaged.
